tests/virtualization/universal/prepare_guests.pm

```diff
@@ -60,6 +59,24 @@ sub create_profile {
     return autoinst_url . "/files/$vm_name.xml";
 }
 
+sub gen_osinfo {
+    my ($vm_name) = @_;
+    my $h_version = get_var("VERSION") =~ s/-SP/./r;
+    my $g_version = $vm_name =~ /sp/ ? $vm_name =~ s/\D*(\d+)sp(\d)\D*/$1.$2/r : $vm_name =~ s/\D*(\d+)\D*/$1/r;
```

Comment:
There are is_sle() function in version_utils and guest_is_sle() function in lib/virt_autotest/vi utils.pm. They may make it easier to check the host & guest versions. It is ok since you have already make it within the gen_isinfo function.